fvm/src/call_manager/default.rs

```diff
@@ -86,14 +92,59 @@ where
     where
         K: Kernel<CallManager = Self>,
     {
+        let mut trace = ExecutionTrace::new(
+            Address::new_id(from),
+            to,
+            method,
+            params.clone(),
+            value.clone(),
+        );
+
         if self.call_stack_depth > self.machine.config().max_call_depth {
+            if self.machine.config().debug {
+                self.exec_trace.as_mut().unwrap().subcalls.push(trace);
+            }
+
             return Err(
                 syscall_error!(LimitExceeded, "message execution exceeds call depth").into(),
             );
         }
         self.call_stack_depth += 1;
         let result = self.send_unchecked::<K>(from, to, method, params, value);
         self.call_stack_depth -= 1;
+
+        trace.msg_receipt = match result {
+            Ok(ref i) => Receipt {
+                exit_code: i.exit_code(),
+                return_data: i.return_data(),
+                gas_used: 0,
+            },
+            Err(ref e) => match e {
+                ExecutionError::OutOfGas => Receipt {
+                    exit_code: ExitCode::SysErrOutOfGas,
+                    return_data: Default::default(),
+                    gas_used: 0,
+                },
+                ExecutionError::Syscall(s) => Receipt {
+                    exit_code: match s.1.try_into() {
+                        Ok(e) => e,
+                        _ => ErrPlaceholder,
+                    },
```

Comment:
Hm. Any chance we could change the format? We're intentionally distinguishing between syscall errors and exit codes.

Reply:
Basically, receipts are really for recording results on-chain, but aren't the best for debugging. In this case, it's also causing us to drop the error message.